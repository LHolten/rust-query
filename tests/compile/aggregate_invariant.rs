use rust_query::{Database, aggregate, migration::schema};

#[schema(Schema)]
pub mod vN {
    pub struct MyTable {
        pub score: i64,
    }
}
use v0::*;

fn test(db: Database<Schema>) {
    db.transaction(|txn| {
        let total = txn.query(|rows| {
            let item = rows.join(MyTable);

            txn.query_one(aggregate(|rows| rows.sum(&item.score)))
        });

        println!("{total}");
    })
}

fn main() {}
