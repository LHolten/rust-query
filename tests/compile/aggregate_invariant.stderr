error: lifetime may not live long enough
  --> tests/compile/aggregate_invariant.rs:16:44
   |
12 |     db.transaction(|txn| {
   |                     --- lifetime `'2` appears in the type of `txn`
13 |         let total = txn.query(|rows| {
   |                                ---- has type `&mut Query<'_, '1, v0::Schema>`
...
16 |             txn.query_one(aggregate(|rows| rows.sum(&item.score)))
   |                                            ^^^^^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'2`
